sources/niLang/src/DataTablePath.cpp

```diff
@@ -52,37 +55,56 @@ iRegex* __stdcall cDataTablePathOp::GetRegex() const
 }
 
 ///////////////////////////////////////////////
-const cString& __stdcall cDataTablePathOp::GetValue() const
-{
+const cString &__stdcall cDataTablePathOp::GetValue() const {
   return mstrValue;
 }
 
+///////////////////////////////////////////////
+const cString& __stdcall cDataTablePathOp::GetPredicate() const
+{
+  return mstrPredicate;
+}
+
 
 //////////////////////////////////////////////////////////////////////////////////////////////
 struct cDataTablePathParserTokenizer : public cIUnknownImpl<iStringTokenizer>
 {
-  cDataTablePathParserTokenizer() : mbFirstChar(eTrue) {}
+  cDataTablePathParserTokenizer() {}
 
   eStringTokenizerCharType __stdcall GetCharType(tU32 c) {
     eStringTokenizerCharType tokType = eStringTokenizerCharType_Normal;
-    if (c == '/' || c == '\\') {
-      tokType = mbFirstChar?eStringTokenizerCharType_SplitterAndToken:eStringTokenizerCharType_Splitter;
+    // slurp [predicate]
+    if (mInPredicate > 0) {
+      if (c == '[') {
+        ++mInPredicate;
+      } else if (c == ']') {
+        --mInPredicate;
+        if (mInPredicate == 0) {
+          tokType = eStringTokenizerCharType_SplitterAndToken;
+        }
+      }
     }
-    else if (c == '@' || c == ',') {
-      tokType = eStringTokenizerCharType_SplitterAndToken;
+    // handle regular tokens
+    else {
+      if (c == '/' || c == '\\' ||  c == '@' || c == ',') {
+        tokType = eStringTokenizerCharType_SplitterAndToken;
```

Comment:
I see that mbFirstChar has been removed. I assume now that the parser handles all the tokens rather than relying on the Splitter variant that made them disappear.

Reply:
yes, I needed the token visible in ParseDataTablePathOp to avoid confusion if someone writes /MyChild/[0]/ and the '/' isn't visible, then you have ambiguity between /MyChild/[0] and /MyChild[0]